packages/framework/src/Framework/Features/Navigation/NavigationMenu.php

```diff
@@ -4,71 +4,22 @@
 
 namespace Hyde\Framework\Features\Navigation;
 
-use Hyde\Facades\Config;
-use Hyde\Support\Models\Route;
-use Hyde\Pages\DocumentationPage;
-use BadMethodCallException;
+use Illuminate\Support\Collection;
+use Illuminate\Contracts\Support\Arrayable;
 
-class NavigationMenu extends BaseNavigationMenu
+class NavigationMenu
 {
-    protected function generate(): void
-    {
-        parent::generate();
-
-        if ($this->dropdownsEnabled()) {
-            $this->moveGroupedItemsIntoDropdowns();
-        }
-    }
-
-    public function hasDropdowns(): bool
-    {
-        return $this->dropdownsEnabled() && count($this->getDropdowns()) >= 1;
-    }
-
-    /** @return array<string, DropdownNavItem> */
-    public function getDropdowns(): array
-    {
-        if (! $this->dropdownsEnabled()) {
-            throw new BadMethodCallException('Dropdowns are not enabled. Enable it by setting `hyde.navigation.subdirectories` to `dropdown`.');
-        }
-
-        return $this->items->filter(function (NavItem $item): bool {
-            return $item instanceof DropdownNavItem;
-        })->values()->all();
-    }
-
-    protected function moveGroupedItemsIntoDropdowns(): void
-    {
-        $dropdowns = [];
-
-        foreach ($this->items as $key => $item) {
-            if ($this->canAddItemToDropdown($item)) {
-                // Buffer the item in the dropdowns array
-                $dropdowns[$item->getGroup()][] = $item;
-
-                // Remove the item from the main items collection
-                $this->items->forget($key);
-            }
-        }
-
-        foreach ($dropdowns as $group => $items) {
-            // Create a new dropdown item containing the buffered items
-            $this->items->add(new DropdownNavItem($group, $items));
-        }
-    }
-
-    protected function canAddRoute(Route $route): bool
-    {
-        return parent::canAddRoute($route) && (! $route->getPage() instanceof DocumentationPage || $route->is(DocumentationPage::homeRouteName()));
-    }
+    /** @var \Illuminate\Support\Collection<string, \Hyde\Framework\Features\Navigation\NavItem> */
```

Comment:
The string here has been used to determine identity to remove duplicates. We don't need that here. Duplicates are already removed in the factory, and if a user pushes items to here we have no call to remove them. We could remove this entirely, or we can solve https://github.com/hydephp/develop/commit/238dac75d43f1192079d826eafb0d7778484bb1d by using a keyed priority (we might then need a check to shift priorities in case of duplicates, which may just add more complexity...)

Reply:
Removed in 95b395ce69a1b3e0e373285adcb64aa0f3cc57d5/088d5b5e6f55d61050890c2cbe4bab312fa62bb6